### PR TITLE
Allow running outside of pull request event context

### PR DESCRIPTION
**Pull request by mcky**
I'd like to be able to run this action on the `deployment_status` event instead of on updates to a pull request to avoid constructing URLs from the PR number. When running it against `deployment_status` there is no `context.payload.pull_request` object, resulting in the following error. 
```
Run ivangabriele/find-and-replace-pull-request-body@v1.1.5
Error: Cannot read properties of undefined (reading 'number')
```

In our own actions code we have some scripts to look up the PR based on the ref, so ideally we'd have something along the lines of this PR to enable explicitly passing a PR number. 

I'm happy to change this implementation as you see fit (and add tests). Or if you'd prefer not to bring this functionality into yours we could fork and rename it, or bring a minimal version of the code into our codebase

### CODE REVIEW

File: index.js
Comment (ivangabriele) [inline index.js:58]: I'm not a big fan of mutable variables, I prefer the code to be as functional and immutable as possible (using `const`) but it's not a deal-breaker for this PR.

**Review by ivangabriele — CHANGES_REQUESTED** [review]
Thank you for this PR @mcky and sorry for not replying sooner, I simply didn't see this PR in my notifications :X.

It would be nice if you could also update the README :). Otherwise that looks good to me.

**Comment by JReinhold** [comment]
I can confirm this works for `workflow_dispatch` events as well. Great job.

https://github.com/storybookjs/storybook/pull/23508/commits/cf2bb2c1b7808ec1c8451586fa0d35bcb723b039

**Comment by ivangabriele** [comment]
@JReinhold I will finalize this PR and merge it tomorrow then ^^.

File: index.js
Comment (mcky) [reply index.js:58]: I also prefer a functional style, although in this case I’d say it’s more of a conditional assignment rather than mutation, as it’s never mutated after being set and both values are primitives 

If we want to avoid making the API call on L65 unless necessary I think it means either:
- keeping it how it is
- Moving the logic into a function like `getPullRequestDetails` that returns early to avoid mutation
- encapsulating the logic that follows it in a function, then calling it from each of the 2 code branches


Im happy to do a bit of refactoring and break it all up into functions, but I didn’t want to step on any toes and tried to make the PR diff as minimal as possible

**Comment by mcky** [comment]
> Thank you for this PR @mcky and sorry for not replying sooner, I simply didn't see this PR in my notifications :X.
> 
> It would be nice if you could also update the README :). Otherwise that looks good to me.

Sweet, I’ll get the README updated and check what I can add to the test suite later today

**Comment by mcky** [comment]
README updated, tests I left because they have some side effects so it'd require quite a bit of change to add additional without changing the source (see my other reply on the thread)

File: index.js
Comment (mcky) [reply index.js:58]: @ivangabriele I had a quick go at refactoring but kept it on a separate branch to not interfere with this PR unless you really want the changes

https://github.com/mcky/find-and-replace-pull-request-body/commit/8f15c80116ceebcaedd5e1740b73ebdc155ae015#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346

It splits it out into a collection of functions, which means it can be unit tested and it solves the mutation thing you mentioned. The current tests suffer from some mutation (`INPUT`, the `context` assignment) as well as `run()` being called at top level in the source file, so splitting out makes it a bit easier to test

_however_ I kind of feel like "if it ain't broke don't fix it" and the refactor doesn't really bring any benefit to end users, so questionable how worthwhile it is

Comment (ivangabriele) [reply index.js:58]: @mcky No you did a nice job, that's definitely cleaner! You can update your PR here. ~I will update the README on my side.~ (you did it already)

**Comment by ivangabriele** [comment]
> README updated, tests I left because they have some side effects so it'd require quite a bit of change to add additional without changing the source (see my other reply on the thread)

As for the tests I will add some once I've merged your updated PR before releasing a new version so the code will be a bit safer.

File: index.js
Comment (mcky) [reply index.js:58]: @ivangabriele I won’t be able to merge the refactor until Monday (as I want to double check it actually works in production) so I can raise that as a follow-up PR to not block this one if you prefer?

or I can merge it with the caveat it hasn’t been properly tested (manually or programmatically) if you’d want to handle that - I don’t want to potentially break things for the sake of style though 😅

Comment (ivangabriele) [reply index.js:58]: I understand! What I'll do then is merging this PR right now and add a few more tests before releasing ^^

**Review by ivangabriele — APPROVED** [review]

**Comment by JReinhold** [comment]
@ivangabriele this hasn't been "released" yet, has it?

**Pull request merged**
